source/common/grpc/google_async_client_impl.cc

```diff
@@ -258,12 +258,20 @@ void GoogleAsyncStreamImpl::writeQueued() {
 }
 
 void GoogleAsyncStreamImpl::onCompletedOps() {
-  Thread::LockGuard lock(completed_ops_lock_);
-  while (!completed_ops_.empty()) {
+  // The items in completed_ops_ execute in the order they were originally added to the queue since
+  // both the post callback scheduled by the completionThread and the deferred deletion of the
+  // GoogleAsyncClientThreadLocal happen on the dispatcher thread.
+  std::deque<std::pair<GoogleAsyncTag::Operation, bool>> completed_ops;
+  {
+    Thread::LockGuard lock(completed_ops_lock_);
+    completed_ops = std::move(completed_ops_);
```

Comment:
`completed_opts_.clear()` here?

Reply:
Isn't this implicit in move semantics?

Reply:
This feels like a repeat of: https://github.com/envoyproxy/envoy/pull/14289/files/606f7af97a41567673a049b609df81cb3b7c99a9#r536419940

Added an ASSERT similar to the one in the dispatcher after a similar std::move.

Reply:
Oh right; I remember. OK that sounds fine and faster.